### PR TITLE
Features for chimps

### PR DESCRIPTION
**Pull request by domi41**
@santisiri @bondsbw @vjrj @brylie 

Implements #103

This PR is based on `master`, and was merged without too much friction.
The ids of DOM elements (`titleContent` and `editor`) were left untouched in the end, except for tags. Time to refactor this later, as ids should be unique.

We fixed some issues introduced in recent commits for non-phone users.
We noticed that there are still critical issues remaining, but they're not introduced by these commits.

---

Roughly, the features tested :

- [x] Authoring
    - [x] Ideas without ballots
    - [x] Ideas with ballots
- [x] Voting
    - [x] Receiving votes upon account creation
    - [x] Using votes on ideas
- [x] Debating
    - [x] Using a chronological tree

Delegation is awaiting feedback in #103

### CODE REVIEW

**Comment by domi41** [comment]
Awww... Seems like recent commit 7d9b756a413b7380a3ea47216566112a0e08fa50 created a conflict. I do not really know what I am doing with git and how to solve this now. Help ?

**Comment by domi41** [comment]
Never mind, doing the obvious thing worked. :tada:

File: imports/ui/templates/components/decision/title/title.js [hunk 1]
```diff
@@ -118,17 +118,6 @@ Template.titleContent.helpers({
   declaration() {
     return getTitle(this);
   },
-  editable() {
```
Comment (domi41) [inline imports/ui/templates/components/decision/title/title.js]: Not 100% we did not break anything on mobile here ; anyways this is going to be React soon.

File: imports/ui/templates/components/decision/tag/tag.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ Template.tag.helpers({
 });
 
 Template.tag.events({
-  'click #add-suggested-tag'(event) {
+  'click .add-suggested-tag'(event) {
```
Comment (domi41) [inline imports/ui/templates/components/decision/tag/tag.js]: This has been re-done in React too.

File: imports/ui/templates/components/decision/tag/tag.html [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-  <li class="w-clearfix tag {{authorization}} {{drag nonDraggable}}" id="draggable" value="{{../_id}}">
+  <li class="w-clearfix tag {{authorization}} {{drag nonDraggable}}" value="{{../_id}}">
     <div class="tag-label">
       <img src="{{pathFor route='home'}}images/hashtag.png" class="hash-icon">
       {{#if suggestion}}
         <a href="{{../url}}" class="w-clearfix w-inline-block hash-link {{authorization hover=true}}">
           <div class="hash-tag">{{{../text}}}</div>
         </a>
-        <div id="add-suggested-tag" class="action">
+        <div id="add-suggested-tag-{{../keyword}}" class="add-suggested-tag action">
```
Comment (domi41) [inline imports/ui/templates/components/decision/tag/tag.html]: A unique id reconstructible from the Tag name is required by the step definitions, to target the button and click on it.

File: imports/ui/templates/components/decision/liquid/liquid.js [hunk 4]
```diff
@@ -175,7 +175,7 @@ Template.liquid.onRendered(function render() {
           Session.set(voteId, this.newVote);
         };
 
-        Meteor.clearTimeout(this.timer);
+        //Meteor.clearTimeout(this.timer);
```
Comment (domi41) [inline imports/ui/templates/components/decision/liquid/liquid.js]: The related `this.timer` is commented out above.

File: imports/ui/templates/components/decision/action/action.js [hunk 5]
```diff
@@ -24,8 +24,8 @@ function disableContractExecution() {
     return true;
   } else if (Session.get('draftOptions') && Session.get('contract').ballotEnabled) {
     return true;
-  } else if (!Session.get('rightToVote')) {
-    return true;
+  // } else if (!Session.get('rightToVote')) {
+  //   return true;
```
Comment (domi41) [inline imports/ui/templates/components/decision/action/action.js]: See #200

File: imports/ui/modules/split.js [hunk 6]
```diff
@@ -52,6 +52,10 @@ const _saveSplitSettings = (left, right) => {
 * @summary splits the view of panels based on user preference
 */
 const _splitRender = () => {
+  // fixme @santisiri
+  // This is ran once BEFORE the _setupSplit because agora is what calls it upon its creation,
+  // which is only the case for phones when in editor mode. See contract.html
+  if (Session.get('resizeSplit') === undefined) { return; }
```
Comment (domi41) [inline imports/ui/modules/split.js]: That hotfix was tough to find, and should not be left as-is @santisiri

File: imports/startup/server/accounts/accounts.js [hunk 7]
```diff
@@ -76,5 +76,7 @@ Accounts.onCreateUser((options, user) => {
     }
   }
 
-  return user;
+  // The default onCreateUser copies the profile into the document, so we need to do it too.
+  // http://docs.meteor.com/api/accounts-multi.html#AccountsServer-onCreateUser
+  return _.extend(user, { profile: profile });
```
Comment (domi41) [inline imports/startup/server/accounts/accounts.js]: This was one of the reasons why the profile was empty.

**Comment by domi41** [comment]
This is ready for review ; an early merge (before the major refacto to React, for instance) is recommended, as a test suite such as this one proves very useful during refactorizations.

**Comment by santisiri** [comment]
There's a massive amount of changes here and the codebase has significantly changed on some files. Maybe split the feature related part into on PR and the rest on another one?

**Comment by domi41** [comment]
I'll rebase this on the new master ; it's mostly all for the features, there were some hotfixes that are meaningless now, so I'll just remove them.

**Comment by vjrj** [comment]
Hi there @domi41: I've just discover this PR after coming back to sovereign today. Can you resolve the conflicts? Good job!

**Comment by domi41** [comment]
Hello @vjrj !
This PR is outdated. See #220 ; I'll rebase it.

**Comment by santisiri** [comment]
A lot of conflcts. Likely to be working with very old code by now.

**Pull request closed without merging**